dmscripts/models/queries.py

```diff
@@ -42,6 +42,15 @@ def process_fields(rules, data):
     return data
 
 
+def rename_fields(rules, data):
+    """ Return data after renaming any keys using the renaming rules """
+    for key in data.keys():
+        if key in rules.keys():
+            data[rules[key]] = data.pop(key)
+
+    return data
+
```

Comment:
So the above function does a couple of things that I'd be sure to note, if not be wary of.

First things first I'm not sure `rules` is the best name possible for the `rules` arg. When I read it I sort of expected a dict like:
`{'string_field_name': some_method_to_transform_said_field}`
maybe something like:
`rules = {
'name': str.capitalize
'age': lambda i: 'You are %d years old' % i
}`
What the function is doing is applying a mapping to the existing field names which is a rule in itself i.e. _The rule_ is if the field is in the mapping dict then swap it with the value in that mapping. Or _apply this mapping to these fields_.

Second thing, you might already know this, with dicts you don't need to call keys when iterating over them. Dicts automatically iterate over their keys. 
`[i for i in d.keys()] == [i for i in d] == [i for i in d.__iter__()]`
It's a bit of a weird one tbh because it's a part of Python that doesn't feel very intuitive.

Finally, dicts are mutable. It always feels a bit weird modifying them in place. You're relying on other people realising what you're up to.
So although you're returning, and reassigning to, `data` when you use this function below you can just as easily not. 
That's why `list.extend` and `dict.update` return None

Something that _appears_ to negate the above is:
```
def rename_fields(mapping, data):
    """If the key has an entry in the mapping replace the key with the value in mapping, otherwise skip."""
    return {mapping.get(key, None) or key: value for key, value in data.iteritems()}
```